### PR TITLE
Update RLBase to v0.8

### PR DESCRIPTION
**Pull request by findmyway**
# Changes

## RLBase related

- `TupleSpace` -> `VectSpace`
- `observe` removed
- `render` -> `Base.display`

## Misc

- use `rng` instead of `seed` if possible
- `POMDPs` updated to v0.9
- Traits of `OpenSpielEnv` are updated

## What's missing?

- [ ] The `get_state(env::OpenSpielEnv)` needs to be expanded to several possible kinds of states. Will be added in the next patch release together with some use cases.

### CODE REVIEW

File: src/environments/open_spiel.jl [hunk 1]
```diff
@@ -106,62 +149,42 @@ end
...
-
-function RLBase.get_observation_space(env::OpenSpielEnv{:observation})
-    s = observation_tensor_size(env.game)
-    MultiContinuousSpace(fill(typemin(Float64), s...), fill(typemax(Float64), s...))
+function RLBase.get_legal_actions_mask(env::OpenSpielEnv, player)
+    if DynamicStyle(env) === SIMULTANEOUS && player == convert(Int, OpenSpiel.SIMULTANEOUS_PLAYER)
+        ones(Bool, length(legal_actions(env.state, player)))
+    else
+        convert(Vector{Bool}, legal_actions_mask(env.state, player))
+    end
```
Comment (findmyway) [inline src/environments/open_spiel.jl]: Reimplement to avoid extra allocation when converting.

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ By default, only some basic environments are installed. If you want to use some
 | `AtariEnv` | [ArcadeLearningEnvironment.jl](https://github.com/JuliaReinforcementLearning/ArcadeLearningEnvironment.jl) | Tested only on Linux|
```
Comment (findmyway) [inline README.md]: update the table

File: src/environments/classic_control/cartpole.jl [hunk 3]
```diff
@@ -136,7 +138,7 @@ function plotendofepisode(x, y, d)
     end
     return nothing
 end
-function RLBase.render(env::CartPoleEnv)
+function render(env::CartPoleEnv)
```
Comment (findmyway) [inline src/environments/classic_control/cartpole.jl]: better to implement Base.display

File: src/environments/non_interactive/pendulum.jl [hunk 4]
```diff
@@ -43,7 +41,7 @@ end
 - `mass = 1.0`
 - `step_size = 0.01`
 - `maximum_time = 10.0`
-- `seed = nothing`
+- `rng = nothing`
```
Comment (findmyway) [inline src/environments/non_interactive/pendulum.jl]: GLOBAL_RNG

File: src/environments/open_spiel.jl [hunk 5]
```diff
@@ -17,62 +19,83 @@ import .OpenSpiel:
...
-- `observation_type`::`Union{Symbol,Nothing}`, Supported values are [`:information`](https://github.com/deepmind/open_spiel/blob/1ad92a54f3b800394b2bc7f178ccdff62d8369e1/open_spiel/spiel.h#L342-L367), [`:observation`](https://github.com/deepmind/open_spiel/blob/1ad92a54f3b800394b2bc7f178ccdff62d8369e1/open_spiel/spiel.h#L397-L408) or `nothing`. The default value is `nothing`, which means `:information` if the game ` provides_information_state_tensor`. If not, it means `:observation`.
+- `state_type`::`Union{Symbol,Nothing}`, Supported values are [`:information`](https://github.com/deepmind/open_spiel/blob/1ad92a54f3b800394b2bc7f178ccdff62d8369e1/open_spiel/spiel.h#L342-L367), [`:observation`](https://github.com/deepmind/open_spiel/blob/1ad92a54f3b800394b2bc7f178ccdff62d8369e1/open_spiel/spiel.h#L397-L408) or `nothing`. The default value is `nothing`, which means `:information` if the game ` provides_information_state_tensor`. If not, it means `:observation`.
+- `rng::AbstractRNG`, used to initial the internal `rng`. And the `rng` will only be used if the environment contains chance node, else it is set to `nothing`.
```
Comment (findmyway) [inline src/environments/open_spiel.jl]: Describe how to set seed of inner env

File: src/environments/open_spiel.jl [hunk 6]
```diff
@@ -81,15 +104,35 @@ function OpenSpielEnv(name; seed = nothing, observation_type = nothing, kwargs..
...
+RLBase.InformationStyle(env::OpenSpielEnv{S,Tuple{C,D,I,N,R,U}}) where {S,C,D,I,N,R,U} = I
+RLBase.NumAgentStyle(env::OpenSpielEnv{S,Tuple{C,D,I,N,R,U}}) where {S,C,D,I,N,R,U} = N
+RLBase.RewardStyle(env::OpenSpielEnv{S,Tuple{C,D,I,N,R,U}}) where {S,C,D,I,N,R,U} = R
+RLBase.UtilityStyle(env::OpenSpielEnv{S,Tuple{C,D,I,N,R,U}}) where {S,C,D,I,N,R,U} = U
+
+Base.copy(env::OpenSpielEnv{S,T,ST,G,R}) where {S,T,ST,G,R} =
+    OpenSpielEnv{S,T,ST,G,R}(copy(env.state), env.game, env.rng)
+
+function Base.show(io::IO, env::OpenSpielEnv)
+    RLBase.print_traits(io, env)
```
Comment (findmyway) [inline src/environments/open_spiel.jl]: no print_trait

File: test/open_spiel.jl [hunk 7]
```diff
@@ -5,34 +5,31 @@
         "kuhn_poker",
         "goofspiel(imp_info=True,num_cards=4,points_order=descending)",
```
Comment (findmyway) [inline test/open_spiel.jl:6]: test seed

**Comment by findmyway** [comment]
@zsunberg I like the new API in POMDPs@v0.9. Much easier to understand and use. 🚀 🚀 🚀

**Comment by zsunberg** [comment]
@findmyway Yes, in POMDPs 0.8 we ended up with something that was very difficult to understand. Hopefully it will be simpler going forward!

**Pull request merged**